Sources/Diffing.swift

```diff
@@ -40,3 +41,118 @@ public extension DiffableViewModel {
         return String(describing: Self.self)
     }
 }
+
+/// MARK: - DifferenceKit Helpers
+
+/// Creates a type-erased Differentiable for DiffableViewModel.
+/// These are only created internally from either `TableCellViewModel` or `CollectionCellViewModel`,
+/// so that we can safely force cast the models back to those types.
+public struct AnyDiffableViewModel {
+    public let model: DiffableViewModel
+
+    private let isContentEqualTo: (AnyDiffableViewModel) -> Bool
+
+    init(_ model: TableCellViewModel) {
+        self.model = model
+
+        let differenceIdentifier = model.diffingKey
+        /// Only compares diff identifiers. This means we'll never get "reload"-type `Changeset`s
+        self.isContentEqualTo = { source in
+            return differenceIdentifier == source.differenceIdentifier
+        }
+    }
+
+    init(_ model: CollectionCellViewModel) {
+        self.model = model
+
+        let differenceIdentifier = model.diffingKey
+        self.isContentEqualTo = { source in
+            return differenceIdentifier == source.differenceIdentifier
+        }
+    }
+}
+
+extension AnyDiffableViewModel: Differentiable {
+
+    public var differenceIdentifier: DiffingKey {
+        return self.model.diffingKey
+    }
+
+    public func isContentEqual(to source: AnyDiffableViewModel) -> Bool {
+        return self.isContentEqualTo(source)
+    }
+}
+
+/// Wraps a `DifferentiableSection` that is a `DiffableViewModel`, so that
+/// `DifferentalSection`-types don't have to implement `Differentiable`, which would lead to an
+/// ambiguous type check for the correct initializer for `StagedChangeset`
+public struct DifferentiableSectionModel<D: DiffableViewModel & DifferentiableSection> {
+    let model: D
+
+    private let isContentEqualTo: (DifferentiableSectionModel) -> Bool
+
+    init(_ model: D) {
+        self.model = model
+        let differenceIdentifier = model.diffingKey
+        self.isContentEqualTo = { source in
+            return differenceIdentifier == source.differenceIdentifier
+        }
+    }
+}
+
+extension DifferentiableSectionModel: Differentiable {
+
+    public var differenceIdentifier: DiffingKey {
+        return self.model.diffingKey
+    }
+
+    public func isContentEqual(to source: DifferentiableSectionModel) -> Bool {
+        return self.isContentEqualTo(source)
+    }
+}
+
+/// MARK: - DifferenceKit Protocol Conformance
+
+extension TableSectionViewModel: DifferentiableSection {
```

Comment:
@jessesquires `AnyDiffableViewModel` is used to satisfy the `DifferentiableSection` protocol. Because `TableSectionViewModel` is `public`, this protocol implementation must be made public (and cascades back to `AnyDiffableViewModel`)

Reply:
Ah, got it 👍